engine/Cuboid.js

```diff
@@ -0,0 +1,78 @@
+class Cuboid {
+  constructor(width, height, depth, options = {
+    position: [0, 0, 0], rotation: [0, 0, 0]
+  }) {
+    this.width = width;
+    this.height = height;
+    this.depth = depth;
+    this.position = options.position;
+    // this.rotation = options.rotation;
+  }
+
+  at(x, y, z) {
+    this.position = [x, y, z];
+    return this;
+  }
+
+  get vertices() {
+    let cubeVertices = [
+      // vordere Fläche
+      -0.5, -0.5,  0.5,
+      0.5, -0.5,  0.5,
+      0.5,  0.5,  0.5,
+      -0.5,  0.5,  0.5,
+
+      // hintere Fläche
+      -0.5, -0.5, -0.5,
+      -0.5,  0.5, -0.5,
+      0.5,  0.5, -0.5,
+      0.5, -0.5, -0.5,
+
+      // obere Fläche
+      -0.5,  0.5, -0.5,
+      -0.5,  0.5,  0.5,
+      0.5,  0.5,  0.5,
+      0.5,  0.5, -0.5,
+
+      // untere Fläche
+      -0.5, -0.5, -0.5,
+      0.5, -0.5, -0.5,
+      0.5, -0.5,  0.5,
+      -0.5, -0.5,  0.5,
+
+      // rechte Fläche
+      0.5, -0.5, -0.5,
+      0.5,  0.5, -0.5,
+      0.5,  0.5,  0.5,
+      0.5, -0.5,  0.5,
+
+      // linke Fläche
+      -0.5, -0.5, -0.5,
+      -0.5, -0.5,  0.5,
+      -0.5,  0.5,  0.5,
+      -0.5,  0.5, -0.5
+    ];
+
+    // TODO: Matrix operation?
+    cubeVertices.forEach((el, index) => {
```

Comment:
This can also be expressed with the transformation matrix. With this you even block yourself from doing instanced rendering, as every model has different vertices. However, I think due to the time and space limits, we won't be able to do instanced rendering anyway...

Reply:
Like said before: This PR is based on the idea that a mesh knows its absolute coordinates.